src/modules/launcher/item.rs

```diff
@@ -166,8 +166,12 @@ impl ItemButton {
 
         if appearance.show_icons {
             let gtk_image = gtk::Image::new();
-            let image =
-                ImageProvider::parse(&item.app_id.clone(), icon_theme, true, appearance.icon_size);
+            let input = if item.app_id.is_empty() {
+                item.name.clone()
+            } else {
+                item.app_id.clone()
+            };
```

Comment:
I tried changed your suggestion by changing `cloned()` to `clone()` but I get:
```
no method named `unwrap_or_else` found for struct `std::string::String` in the current scope
method not found in `String`
```
Best I seem to be able to do (without failing clippy) is:

```suggestion
            let input = item
                .app_id
                .is_empty()
                .then(|| item.name.clone())
                .unwrap_or_else(|| item.app_id.clone());
```

I think the if is clearer but will defer to whatever you prefer - not familiar with the rest of the code. The issue is app_id is not None but the empty string in this case.

Reply:
Oh sorry, I misread it and thought it was an `Option` not an empty string. In that case, I think it's best as you originally had it.